public/docs/_examples/toh-5/dart/lib/app_component.dart

```diff
@@ -0,0 +1,56 @@
+// #docplaster
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'package:angular2_tour_of_heroes/heroes_component.dart';
+import 'package:angular2_tour_of_heroes/hero_service.dart';
+import 'package:angular2/router.dart';
+import 'package:angular2_tour_of_heroes/dashboard_component.dart';
+// #docregion hero-detail-import
+import 'package:angular2_tour_of_heroes/hero_detail_component.dart';
+// #enddocregion hero-detail-import
+
+@Component(
+    selector: 'my-app',
+// #docregion template
+    template: '''
+<h1>{{title}}</h1>
+<nav>
+  <a [routerLink]="['Dashboard']">Dashboard</a>
+  <a [routerLink]="['Heroes']">Heroes</a>
+</nav>
+<router-outlet></router-outlet>
+''',
+// #enddocregion template
+// #docregion style-urls
+    styleUrls: const ['app_component.css'],
+// #enddocregion style-urls
+    directives: const [ROUTER_DIRECTIVES],
```

Comment:
This can also be done in TypeScript (see https://github.com/angular/angular/issues/5938 for instructions). I guess this would be great to show in the tutorial (TS and Dart). Adding `ROUTER_DIRECTIVES` to each component is error prone and cumbersome.